### PR TITLE
feat(crypto): adding ed25519 support

### LINKED ISSUE #15: ED25519 support

### PR DESCRIPTION
**Pull request by kehiy**
- Fixes #15

### CODE REVIEW

File: examples/example_key_generation.py [hunk 1]
```diff
@@ -12,9 +14,9 @@ def main() -> None:
     parser.add_argument(
         "--address-type",
         type=int,
-        choices=[1, 2],
+        choices=[1, 2, 3],
         default=2,
-        help="Type of address to generate: 1 for Validator, 2 for BLSAccount (default: 2)",
+        help="Type of address to generate: 1 for Validator, 2 for BLSAccount, 3 for ED25519Account (default: 2)",
```
Comment (b00f) [inline examples/example_key_generation.py:19]: E**d**25519Account

File: examples/example_key_generation.py [hunk 2]
```diff
@@ -2,8 +2,10 @@
 import secrets
 
 from pactus.crypto import CryptoConfig
-from pactus.crypto.address import AddressType, Address
-from pactus.crypto.bls.private_key import PrivateKey, PublicKey
+from pactus.crypto.address import AddressType
+from pactus.crypto.bls.private_key import PrivateKey
+
+from pactus.crypto.ed25519.private_key import PrivateKey as Edprivk
```
Comment (b00f) [inline examples/example_key_generation.py:8]: EdPrivKey

**Review by b00f — CHANGES_REQUESTED** [review]

File: examples/example_key_generation.py [hunk 3]
```diff
@@ -31,22 +32,29 @@ def main() -> None:
...
+            pub = sec.public_key()
+            addr = pub.account_address()
+            show(sec, pub, addr)
+
         case _:
             return
 
 
-def show(sec: PrivateKey, pub: PublicKey, addr: Address):
+def show(sec: any, pub: any, addr: any):
```
Comment (b00f) [inline examples/example_key_generation.py:57]: Why `any`? Still we can keep the interfaces.

File: pactus/crypto/ed25519/private_key.py [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+            msg = "Private key data must be 32 bytes long"
+            raise ValueError(msg)
+
+        scalar = ed25519.Ed25519PrivateKey.from_private_bytes(bytes(data))
+        return cls(scalar)
+
+    def raw_bytes(self) -> bytes:
+        return self.scalar.private_bytes_raw()
+
+    def string(self) -> str:
```
Comment (b00f) [inline pactus/crypto/ed25519/private_key.py:47]: define as `classmethod`

File: pactus/crypto/ed25519/private_key.py [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+            raise ValueError(msg)
+
+        if len(data) != PRIVATE_KEY_SIZE:
+            msg = "Private key data must be 32 bytes long"
+            raise ValueError(msg)
+
+        scalar = ed25519.Ed25519PrivateKey.from_private_bytes(bytes(data))
+        return cls(scalar)
+
+    def raw_bytes(self) -> bytes:
```
Comment (b00f) [inline pactus/crypto/ed25519/private_key.py:44]: Same here

**Comment by kehiy** [comment]
@b00f Class method is not used/not used same as BLS example. As I know they need self and they can't be class method.

File: pactus/crypto/ed25519/ruff.toml [hunk 6]
```diff
@@ -0,0 +1,2 @@
+[lint]
+ignore = ["E722"]
```
Comment (b00f) [inline pactus/crypto/ed25519/ruff.toml:2]: Why do we need to ignore it?

**Review by b00f — APPROVED** [review]

**Pull request merged**